posthog/models/team/team.py

```diff
@@ -280,3 +281,68 @@ def get_available_features_for_team(team_id: int):
     )
 
     return available_features
+
+
+def get_effective_membership_level(team_id: int, user_id: int) -> Optional["OrganizationMembership.Level"]:
+    """
+    Given a team_id and a user_id, return an effective membership level.
+
+    None returned if the user has no explicit membership and organization access is too low for implicit membership.
+    """
+    from posthog.models.organization import OrganizationMembership
+
+    try:
+        (membership_id, available_features, access_control, level) = (
+            OrganizationMembership.objects.filter(organization__team=team_id, user_id=user_id)
+            .values_list(
+                "id",
+                "organization__available_features",
+                "organization__team__access_control",
+                "level",
+            )
+            .get()
+        )
+    except OrganizationMembership.DoesNotExist:
+        return None
+
+    return get_effective_membership_level_for_parent_membership(
+        membership_id=membership_id,
+        team_id=team_id,
+        available_features=available_features,
+        access_control=access_control,
+        level=level,
+    )
+
+
+def get_effective_membership_level_for_parent_membership(
```

Comment:
This looks like a lot of duplicated code, what's the plan for it?

Reply:
Move everything over to the new functions. They should be identical (that's not for certain though, more an aspiration) in functionality except for the signature only relying on ids rather than Django objects.